x11-packages/xwayland/xorg-server-1.20.1_os_utils.c.patch

```diff
@@ -19,20 +18,7 @@ diff -uNr xorg-server-1.20.1/os/utils.c xorg-server-1.20.1.mod/os/utils.c
          if (haslock) {
              /*
               * We're done.
-@@ -1383,11 +1383,7 @@
-         p = -1;
-         break;
-     case 0:                    /* child */
--        if (setgid(getgid()) == -1)
--            _exit(127);
--        if (setuid(getuid()) == -1)
--            _exit(127);
--        execl("/bin/sh", "sh", "-c", command, (char *) NULL);
-+        execl("@TERMUX_PREFIX@/bin/sh", "sh", "-c", command, (char *) NULL);
-         _exit(127);
-     default:                   /* parent */
-         do {
-@@ -1453,10 +1449,6 @@
```

Comment:
Removing this part of patch is not an option because:
1. Setuid and setgid are not supported in termux.
2. Running /bin/sh and not @TERMUX_PREFIX@/bin/sh is not supported.

Reply:
![IMG_20240527_235518](https://github.com/termux/termux-packages/assets/59302530/375b735e-64b2-4ca5-be62-2faf24eb7226)

Reply:
Probably this screenshot should be followed by some description...

Reply:
Explanation, `termux-exec` is shimming the regular shebang to work.
https://github.com/termux/termux-exec

Which is not a valid solution at build time.

Reply:
`setuid`,`setgid`,`/bin/sh` only need to patch the functions `Popen` and `Fopen` in [xwayland-24.1/os/utils.c](https://gitlab.freedesktop.org/xorg/xserver/-/blob/xwayland-24.1/os/utils.c?ref_type=heads#L1304)

https://github.com/tqfx/termux-packages/blob/xwayland/x11-packages/xwayland/xorg-server-1.20.1_os_utils.c.patch

Reply:
https://gitlab.freedesktop.org/xorg/xserver/-/compare/xwayland-23.2...xwayland-24.1.0?from_project_id=371&page=10&straight=true#cccacbd9d6b73bb5139f8457283f61747171576e